scripts/mutect2_wdl/mutect2.wdl

```diff
@@ -232,85 +215,60 @@ task M2 {
       GATK_JAR=${gatk4_jar_override}
   fi
 
+  java -Xmx4g -jar $GATK_JAR GetSampleName -I ${tumor_bam} -O tumor_name.txt
+  output_vcf=`cat tumor_name.txt`-tumor-only.vcf
+
+   # We need to create a file regardless, even if it stays empty
+   touch bamout.bam
+
   if [[ "_${normal_bam}" == *.bam ]]; then
       java -Xmx4g -jar $GATK_JAR GetSampleName -I ${normal_bam} -O normal_name.txt
       normal_command_line="-I ${normal_bam} -normal `cat normal_name.txt`"
-  else
-      # Note that normal_name.txt is always created, it's just empty if no normal sample was given.
-      #     This is done to allow an output parameter of the normal sample.
-      touch normal_name.txt
+      output_vcf=`cat tumor_name.txt`-vs-`cat normal_name.txt`.vcf
   fi
 
-  java -Xmx4g -jar $GATK_JAR GetSampleName -I ${tumor_bam} -O tumor_name.txt
 
-  # We need to create a file regardless, even if it stays empty
-  touch bamout.bam
 
   java -Xmx4g -jar $GATK_JAR Mutect2 \
     -R ${ref_fasta} \
     -I ${tumor_bam} \
     -tumor `cat tumor_name.txt` \
     $normal_command_line \
-    ${"--germline_resource " + gnomad} \
-    ${"--normal_panel " + pon} \
+    ${"--germline-resource " + gnomad} \
+    ${"-pon " + pon} \
     ${"-L " + intervals} \
-    -O "${output_vcf_name}.vcf" \
+    -O $output_vcf \
     ${true='--bamOutput bamout.bam' false='' is_bamOut} \
     ${m2_extra_args}
   >>>
 
   runtime {
-      docker: "${m2_docker}"
-      memory: "5 GB"
-      disks: "local-disk " + 500 + " HDD"
-      preemptible: "${preemptible_attempts}"
+        docker: "${gatk_docker}"
+        memory: select_first([mem, 5]) + " GB"
+        disks: "local-disk " + select_first([disk_space_gb, 100]) + " HDD"
+        preemptible: select_first([preemptible_attempts, 2])
   }
 
   output {
-    File output_vcf = "${output_vcf_name}.vcf"
+    File output_vcf = glob("*.vcf")[0]
```

Comment:
This seems a bit dangerous/fragile......

Reply:
fixed